src/aotcompile.cpp

```diff
@@ -462,9 +464,12 @@ void *jl_create_native_impl(jl_array_t *methods, LLVMOrcThreadSafeModuleRef llvm
     }
 
     data->M = std::move(clone);
-    if (!ct->reentrant_timing-- && measure_compile_time_enabled) {
-        auto end = jl_hrtime();
-        jl_atomic_fetch_add_relaxed(&jl_cumulative_compile_time, end - compiler_start_time);
+    if (timed) {
+        if (measure_compile_time_enabled) {
+            auto end = jl_hrtime();
+            jl_atomic_fetch_add_relaxed(&jl_cumulative_compile_time, end - compiler_start_time);
+        }
+        ct->reentrant_timing ^= 1;
```

Comment:
minor nit: the usual pattern for this expression is to make a signed mask value as such:
```suggestion
        ct->reentrant_timing &= ~1;
```

Reply:
The sign extension is a neat trick, I'm just wary of signed bitmasks in general because occasionally the sign bit is set to 0 in the mask, which breaks it slightly. I'll go back to the conventional clear operation though.